public/app/features/explore/TraceView/createSpanLink.tsx

```diff
@@ -30,11 +32,11 @@ export function createSpanLinkFactory(splitOpenFn: (options: { datasourceUid: st
     // it manually here instead of leaving it for the data source to supply the config.
 
     const dataLink: DataLink<LokiQuery> = {
-      title: lokiDs.name,
+      title: dataSourceSettings.name,
       url: '',
       internal: {
-        datasourceUid: lokiDs.uid,
-        datasourceName: lokiDs.name,
+        datasourceUid: dataSourceSettings.uid,
+        datasourceName: dataSourceSettings.name,
```

Comment:
We should move this into the datasource and add the links on the dataFrame. That way the data source controls this which is aligned with how other links work and also means that `get<some_datasource>QueryFromSpan` will live in tracing datasource which also makes more sense because you can do different mapping per DS (like we could implement something for x-ray and cloudwatch theoretically)

Reply:
This is something that we will try to do in a later pr.